### PR TITLE
Homework8

### PR DESCRIPTION
**Pull request by OlegShvets1**
Homework8

### CODE REVIEW

File: src/Lesson_8/Task_2/Main_2.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        int a = 5;
+        int b = 0;
+        try {
+            return a / b;
+        } catch (ArithmeticException e) {
+
+            System.out.println("Divided by zero operation cannot possible");
+        }
+
+        return a;
```
Comment (Evangard) [inline src/Lesson_8/Task_2/Main_2.java:34]: You don't need to return a at the end. You need to throw new exception as you do above but with your own message.

File: src/Lesson_8/Task_3/Main_3.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        System.out.println(a);
+
+        try {
+            int b = 10 / a;
+            int[] c = {1};
+            System.out.println(c[1]);
+        } catch (ArithmeticException | ArrayIndexOutOfBoundsException e) {
+            System.out.println("Divided by zero operation cannot possible");
+            System.out.println("Element in the array was not found");
+        }
```
Comment (Evangard) [inline src/Lesson_8/Task_3/Main_3.java:31]: This is not correct, as system returns two messages "Divided by zero operation cannot possible" and "Element in the array was not found" if you do dividing by 0. You need to write two separate catch here. One for ArithmeticException and another one for ArrayIndexOutOfBoundsException. And you have to print a cause, not your own message - e.getCause();

File: src/Lesson_8/Task_4/Task_4.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}*/
+
+public class Task_4 {
+    public static void main(String[] args) {
+        try {
+            int a = 0;
+            int b = 4 / a;
+        } catch (ArithmeticException e) {
+            e.printStackTrace();
+        }
```
Comment (Evangard) [inline src/Lesson_8/Task_4/Task_4.java:22]: In this case system works fine but indeed you just need to swap Exception and ArithmeticException. Remember the order of exceptions. Parent should be at the end.

File: src/Lesson_8/Task_2/Main_2.java (hunk 1 above)
Comment (OlegShvets1) [reply src/Lesson_8/Task_2/Main_2.java:34]: Corrected

File: src/Lesson_8/Task_3/Main_3.java (hunk 2 above)
Comment (OlegShvets1) [reply src/Lesson_8/Task_3/Main_3.java:31]: Corrected

File: src/Lesson_8/Task_4/Task_4.java (hunk 3 above)
Comment (OlegShvets1) [reply src/Lesson_8/Task_4/Task_4.java]: Corrected

**Review by Evangard — APPROVED** [review]
You can merge it.

**Pull request merged**